### PR TITLE
CI: parallel-safe network system test

### PR DESCRIPTION
**Pull request by edsantiago**
- replace random_string with safename in container/network names
- add ci:parallel tags where possible.
  - where not possible, add explanations

Signed-off-by: Ed Santiago <santiago@redhat.com>
```release-note
None
```

### CODE REVIEW

File: test/system/500-networking.bats
Comment (Luap99) [inline test/system/500-networking.bats:982]: I wonder how much value this check here does. I mean I want to check that we do no leak the netns files however we only do so here in a ugly way IMO. Podman uses a consistent netns file name, as such I wonder if we should just add such check in the leak_check  at the end of the suite. Then it will cover all tests not just this one. There should never be reason for us to leak the file, however it would be hard to know where it was leakded in case of failures but I guess if leaks happen we can instrument teardown to ls /run/netns to notice where it starts leaking...

Of course out of scope of this PR.

**Review by Luap99 — APPROVED** [review]
LGTM

File: test/system/500-networking.bats
Comment (edsantiago) [reply test/system/500-networking.bats:982]: Well.... this comment does have your name on it https://github.com/containers/podman/blob/ae4eaaf1ae6921eab9d5ba5f3c9252cb267e77b1/test/system/500-networking.bats#L1002-L1003

I'm all for adding a leak check, but what would that look like? Right now on my laptop I see six `/run/netns/netns*` files with no tests running. Looks like I have some orphan pasta processes

Comment (Luap99) [reply test/system/500-networking.bats:982]: I know this was added by me, my comment was more of if it is blocking parallel tests it may not be a good place because checking leaks in one place doesn't help much if we still leak in other places 

> I'm all for adding a leak check, but what would that look like? Right now on my laptop I see six /run/netns/netns* files with no tests running. Looks like I have some orphan pasta processes

pasta should not be used as root so I don't think they relate to each other. If there are leaks right now then we need to debug/fix them first of course. Anyhow as said not in this PR. I will file a new issue to consider this.

File: test/system/500-networking.bats
Comment (edsantiago) [inline test/system/500-networking.bats:959]: This fixes a leak. Found by a leak check I've added (experimental) to #23275.

Comment (Luap99) [reply test/system/500-networking.bats:959]: Not blocking but doesn't this mean this always leaks the netns when one of the checks above fails?
Should this be moved into a custom teardown function?

Comment (edsantiago) [reply test/system/500-networking.bats:959]: Maybe, but what's different about here vs all the other places that leak netns on failure? Like [this one](https://api.cirrus-ci.com/v1/artifact/task/4687505503551488/html/sys-podman-fedora-39-root-host-boltdb.log.html#t--00369p) (scroll to bottom): every time `600 shell completion` fails, it leaks four netns, the UUID-named ones, even though the `network create` in that test uses `safename`. It has been bothering me for a while, but I've only spent minimal time investigating. 

Anyhow, if netns cleanup is more important than other container-related cleanup, I'm happy to move that into teardown somehow.

Comment (Luap99) [reply test/system/500-networking.bats:959]: if the netns files created by podman are leaked then this is a podman bug, even in the error paths we should be removing them

I don't think it is super important as it would only leak on error cases and there is no problems with leaks other than a bit of kernel memory keeping it around. And a reboot fixes that so I think that is fine for machines where these tests are run.

**Comment by Luap99** [comment]
/lgtm

Given this is has been up for one week without another review I am going to merge it without a second review

**Pull request merged**
